core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java

```diff
@@ -70,13 +70,13 @@ private OnDemandExtensions() {
     }
 
     @SuppressWarnings("PMD.AvoidRethrowingException")
-    private static Object invoke(Object target, Method method, Object[] args) {
+    private static Object invoke(Object target, Method method, Object[] args) throws Exception {
         try {
             if (Proxy.isProxyClass(target.getClass())) {
                 return Proxy.getInvocationHandler(target).invoke(target, method, args);
             }
             return MethodHandles.lookup().unreflect(method).bindTo(target).invokeWithArguments(args);
-        } catch (RuntimeException | Error e) {
+        } catch (Exception | Error e) {
```

Comment:
Since the method now throws vanilla Exception, is there any downside to just removing the try/catch and letting the error bubble up as is?

Reply:
The one downside I can think of is that it makes this change slightly more breaking, as now we stop wrapping all `Throwable`s in a `RunTimeException`. 

Although, in practice, I've never seen any `Throwable` that isn't an `Exception` or `Error`, and allowing errors to bubble up provides maximal exception transparency. 

I think it's best to rip the bandaid off all at once, so I'll make this change

Reply:
Good catch.  Fewer cases to think of is better, let's remove it.

Reply:
My bad. It actually turns out that any `Throwable` that isn't `Exception` or `Error` needs to be wrapped in a `RunTimeException`. If not, then we need to say `throws Throwable` in the method signature, which in turn creates the need to change `throws Exception` to `throws Throwable` in many other classes, some of which we can't change such as `java.util.concurrent.Callable#call`. Removing it won't be worth it.

Reply:
Ah, of course.  We should have used `X extends Throwable` instead of `X extends Exception` for our exception bounds, but that error can't be fixed until jdbi 4.